google/cloud/pubsub/samples/samples.cc

```diff
@@ -912,137 +896,105 @@ void ValidateAvroSchema(google::cloud::pubsub::SchemaServiceClient client,
   }
   //! [validate-protobuf-schema]
-  (std::move(client), argv.at(0));
+  (std::move(client), argv.at(0), argv.at(1));
 }
 
 void ValidateMessageAvro(google::cloud::pubsub::SchemaServiceClient client,
                          std::vector<std::string> const& argv) {
   //! [validate-message-avro]
   namespace pubsub = ::google::cloud::pubsub;
-  [](pubsub::SchemaServiceClient client, std::string const& project_id) {
-    auto constexpr kDefinition = R"js({
-      "type": "record",
-      "name": "State",
-      "namespace": "google.cloud.pubsub.samples",
-      "doc": "A list of states in the United States of America.",
-      "fields": [
-        {
-          "name": "name",
-          "type": "string",
-          "doc": "The common name of the state."
-        },
-        {
-          "name": "post_abbr",
-          "type": "string",
-          "doc": "The postal code abbreviation of the state."
-        }
-      ]
-    })js";
-    auto constexpr kMessage = R"js({
-          "name": "New York",
-          "post_abbr": "NY"
-    })js";
+  [](pubsub::SchemaServiceClient client, std::string const& project_id,
+     std::string const& schema_definition_file,
+     std::string const& message_file) {
+    auto definition = ReadFile(schema_definition_file);
+    if (!definition) throw std::move(definition).status();
+
+    auto message = ReadFile(message_file);
+    if (!message) throw std::move(message).status();
+
     google::pubsub::v1::ValidateMessageRequest request;
     request.set_parent(google::cloud::Project(project_id).FullName());
     request.mutable_schema()->set_type(google::pubsub::v1::Schema::AVRO);
-    request.mutable_schema()->set_definition(kDefinition);
-    request.set_message(kMessage);
+    request.mutable_schema()->set_definition(definition.value());
+    request.set_message(message.value());
     request.set_encoding(google::pubsub::v1::JSON);
     auto response = client.ValidateMessage(request);
     if (!response) throw std::move(response).status();
     std::cout << "Message is valid\n";
   }
   //! [validate-message-avro]
-  (std::move(client), argv.at(0));
+  (std::move(client), argv.at(0), argv.at(1), argv.at(2));
 }
 
 void ValidateMessageProtobuf(google::cloud::pubsub::SchemaServiceClient client,
                              std::vector<std::string> const& argv) {
   //! [validate-message-protobuf]
   namespace pubsub = ::google::cloud::pubsub;
-  [](pubsub::SchemaServiceClient client, std::string const& project_id) {
+  [](pubsub::SchemaServiceClient client, std::string const& project_id,
+     std::string const& schema_definition_file,
+     std::string const& message_file) {
+    auto definition = ReadFile(schema_definition_file);
+    if (!definition) throw std::move(definition).status();
+
+    auto message_file_contents = ReadFile(message_file);
+    if (!message_file_contents) throw std::move(message_file_contents).status();
     google::cloud::pubsub::samples::State data;
```

Comment:
So this example loads the `.proto` and `.textproto` files, but can only work if both use a `google::cloud::pubsub::samples::State` message?

Seems like we either allow for any message type, or we don't bother with the additional flexibility?